routes/farmer/farmer.js

```diff
@@ -264,12 +257,43 @@ exports.updateFarmById = function(req, res) {
                     res.send({error: "Not Found"});
                 }
             }
+        });*/
+
+
+        Farmer.findById(req.params.id, function (err, doc) {
+
+            var this_farm = doc.fr_farms.id(req.params.farm_id);
+
+            this_farm.fr_name = req.body.fr_name;
+
+            this_farm.di_district = req.body.di_district;
+
+            this_farm.ad_address1 = req.body.ad_address1;
+
+            this_farm.ad_address2 = req.body.ad_address2;
+
+            this_farm.ad_latitude = req.body.ad_latitude;
+
+            this_farm.ad_longitude = req.body.ad_longitude;
+
+            this_farm.ad_city = req.body.ad_city;
+
+            this_farm.ad_country = req.body.ad_country;
+
+            this_farm.ad_longitude = req.body.ad_longitude;
+
+            this_farm.fr_size = req.body.fr_size;
+
+            doc.save();
```

Comment:
@nick22891  Same as above, we need to know the outcome of the update and send the appropriate response to the front end of the application.